lib/buffer.js

```diff
@@ -42,34 +42,92 @@ function alignPool() {
   }
 }
 
-
-function Buffer(arg, encoding) {
+/**
+ * The Buffer() construtor is "soft deprecated" ... that is, it is deprecated
+ * in the documentation and should not be used moving forward. Rather,
+ * developers should use one of the three new factory APIs: Buffer.from(),
+ * Buffer.allocUnsafe() or Buffer.alloc() based on their specific needs. There
+ * is no hard deprecation because of the extent to which the Buffer constructor
+ * is used in the ecosystem currently -- a hard deprecation would introduce too
+ * much breakage at this time. Eventually, however, we should re-evaluate that
+ * choice.
```

Comment:
@jasnell That's a larger discussion that should probably happen after lots more code is using the new factory functions. Otherwise, users who haven't heard of the new methods will see that `Buffer()` zero fills and may write code that's insecure on older versions of Node.